packages/assets-controllers/src/NftController.ts

```diff
@@ -512,6 +512,53 @@ export class NftController extends BaseController<
     });
   }
 
+  /**
+   * Fetches NFT Collection Metadata from the NFT API.
+   *
+   * @param contractAddresses - The contract addresses of the NFTs.
+   * @param chainId - The chain ID of the network where the NFT is located.
+   * @returns NFT collections metadata.
+   */
+  async fetchNftCollectionsMetadata(
+    contractAddresses: string[],
+    chainId: Hex,
+  ): Promise<{
+    collections: Collection[];
+  }> {
+    return await this.#getNftCollectionInformationFromApi(
+      contractAddresses,
+      chainId,
+    );
+  }
+
+  async #getNftCollectionInformationFromApi(
+    contractAddress: string[],
+    chainId: Hex,
+  ) {
+    const urlParams = new URLSearchParams({
+      chainId,
+    });
+
+    for (const address of contractAddress) {
+      urlParams.append('contract', address);
+    }
+
+    return await handleFetch(
+      `${this.#getNftCollectionApi()}?${urlParams.toString()}`,
+      {
+        headers: {
+          Version: '1',
+        },
+      },
+    );
+  }
+
+  #getNftCollectionApi(): string {
+    // TODO: Either fix this lint violation or explain why it's necessary to ignore.
+    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
```

Comment:
What are your thoughts on explaining why this is necessary to ignore?

Note that this lint rule is known to have false negatives, so if `NFT_API_BASE_URL` is really a string, then it might be as simple as saying:

``` suggestion
    // False negative.
    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
```

Alternatively, if `NFT_API_BASE_URL` is a string, is this necessary?

``` suggestion
```

Reply:
I will add the `False negative` here, I still have lint issues even after removing this disabling rule.
Also linter passing locally but failing on CI which I couldn't explain at first look.